syntax = "proto3";
package auth;

//Login message struct
message AuthReq {
    string username = 1;
    string password = 2;
    int64 auth_type = 3;   
}

message AuthRes {
    string access_token = 1;
    int64 expires_in = 2;
    int64 refresh_expires_in = 3;
    string refresh_token = 4;
    string token_type = 5;
    string session_state = 6;
    string scope = 7;
    string hash_key = 8;
}

//Register message struct
message RegisterReq {
    string email = 1;
    string username = 2;
    string password = 3;
    int64 auth_type = 4;
    string first_name = 5;
    string last_name = 6;
}

message RegisterRes {
    bool success = 1;
}

service Auth {
    rpc login(AuthReq) returns (AuthRes) {};
    rpc register(RegisterReq) returns (RegisterRes) {};
}
